src/base.js

```diff
@@ -364,14 +396,59 @@ export default class KintoClientBase {
       return raw ? { json: msg, headers: { get() {} } } : msg;
     }
     await this.fetchServerSettings();
-    const result = await this.http.request(this.remote + request.path, {
-      ...request,
-      body: stringify ? JSON.stringify(request.body) : request.body,
-    });
+    // FIXME: Because any fetch() option is accepted here, `request`
+    // could have lots of parameters that wouldn't be accepted as part
+    // of a batch request. This makes it easier to shoot yourself in
+    // the foot. We should probably only pass through the four
+    // parameters accepted in a batch request: method, body, path, and
+    // headers.
```

Comment:
The natural fix here is to select only those properties from `request` using something like

```
    const result = await this.http.request(
      this.remote + request.path,
      {
        method: request.method,
        headers: request.headers,
        body: stringify ? JSON.stringify(request.body) : request.body,
      },
      {
        retry: this._getRetry(options),
      }
    );
```

But this means that if `request` doesn't provide those things, we'll get `undefined` instead of missing fields. That seems a little ugly to me. What do you think?